scraper/scrape.py

```diff
@@ -214,12 +265,13 @@ def scrape(pages_number):
                 all_product_details.append(details)
 
         print(f"Scraped {len(all_product_details)} products from page {page}.")
+        save_to_json(all_product_details, page)
         for product in all_product_details:
             print(product)
 
         print("\n" + "=" * 40 + "\n")
 
 
 if __name__ == "__main__":
-    NUMBER_OF_PAGES_TO_SCRAPE = 22
+    NUMBER_OF_PAGES_TO_SCRAPE = 21
```

Comment:
Some error handling if the page doesn't exist would be nice, instead of having to manually change it when the source shop's inventory changes 

Or ideally script should accept `number of products` instead of `number of pages` and automatically calculate how many pages to scrap

No matter which option you choose it should be either script argument or entry in the `.env` file

Reply:
IMHO, I cant see any nice solution for this, because the original website doesnt return any error of not existing page, it just redirects to the last one, however we can check if the last product's id is different from the previous page to detect loop.